bindings/pybind11/idyntree_type_caster.h

```diff
@@ -0,0 +1,224 @@
+#ifndef IDYNTREE_PYBIND11_IDYNTREE_TYPE_CASTER_H
+#define IDYNTREE_PYBIND11_IDYNTREE_TYPE_CASTER_H
+
+#include <iostream>
+#include <string>
+#include <type_traits>
+
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/MatrixFixSize.h>
+#include <iDynTree/Core/Rotation.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/VectorFixSize.h>
+
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+
+namespace pybind11
+{
+namespace detail
+{
+template <template <unsigned int...> class base, typename derived>
+struct is_base_of_template_int_impl
+{
+    template <unsigned int... Ts> static constexpr std::true_type test(const base<Ts...>*);
+    static constexpr std::false_type test(...);
+    using type = decltype(test(std::declval<derived*>()));
+};
+
+template <template <unsigned int...> class base, typename derived>
+using is_base_of_template_int = typename is_base_of_template_int_impl<base, derived>::type;
+
+template <typename Type>
+using is_idyntree_vector_fix_size = is_base_of_template_int<iDynTree::VectorFixSize, Type>;
+template <typename Type>
+using is_idyntree_matrix_fix_size = is_base_of_template_int<iDynTree::MatrixFixSize, Type>;
+
+// Type caster for all the classes that inherits from VectorDynSize
+template <typename Type>
+struct type_caster<Type, enable_if_t<std::is_base_of<iDynTree::VectorDynSize, Type>::value>>
+{
+    PYBIND11_TYPE_CASTER(Type, _(PYBIND11_TOSTRING(Type)));
+
+    /**
+     * Conversion from Python to C++
+     */
+    bool load(handle src, bool convert)
+    {
+        if (!convert && array_t<double>::check_(src))
+        {
+            return false;
+        }
+
+        auto buf
+            = pybind11::array_t<double,
+                                pybind11::array::c_style | pybind11::array::forcecast>::ensure(src);
+        if (!buf)
+            return false;
+
+        // the number of dimension must be equal to 1 since it is a vector
+        const auto dims = buf.ndim();
+        if (dims != 1)
+            return false;
+
+        // store the data
+        value = Type(buf.data(), buf.size());
+        return true;
+    }
+
+    /**
+     * Conversion from C++ to Python
+     */
+    static handle cast(Type src, return_value_policy policy, handle parent)
```

Comment:
`return_value_policy policy` this partially answer my previous question, at least for the C++->Python side.

This side is good. We copy the object into Python